viewer/packages/tools/src/SmartOverlay/SmartOverlayTool.ts

```diff
@@ -80,121 +77,59 @@ export class SmartOverlayTool extends Cognite3DViewerToolBase {
     this._textOverlay = this.createTextOverlay('', this._defaultTextOverlayToCursorOffset);
     viewer.domElement.appendChild(this._textOverlay);
 
-    this._singeOverlayCollection = new Overlay3DCollection(undefined, {
-      maxPointSize: toolParameters?.maxPointSize
-    });
-    this._overlayPoints.push(this._singeOverlayCollection);
-    viewer.addObject3D(this._singeOverlayCollection);
-
     viewer.canvas.addEventListener('mousemove', this.onMouseMove);
 
     viewer.on('click', this.onMouseClick);
```

Comment:
I suppose this is fine for testing, but in the end, the callbacks shouldn't be added here, but registered by the user, right?

Reply:
So, in the same way as we subscribe for hover/click events of html element, we subscribe to events of all overlays created with one tool.